packages/gafl-webapp-service/src/pages/contact/name/__tests__/update-transaction.spec.js

```diff
@@ -0,0 +1,55 @@
+import { NAME } from '../../../../uri.js'
+import updateTransaction from '../update-transaction.js'
+
+const createRequestMock = (pageData, licenseeData, setCurrentPermission = jest.fn()) => ({
+  cache: () => ({
+    helpers: {
+      page: {
+        getCurrentPermission: pageData
+      },
+      transaction: {
+        getCurrentPermission: () => licenseeData,
+        setCurrentPermission
+      }
+    }
+  })
+})
+
+describe('update-transaction', () => {
+  beforeEach(jest.clearAllMocks)
+  it('gets the payload from the name page cache', async () => {
+    const samplePayload = { payload: { 'first-name': 'firstName', 'last-name': 'lastName' } }
+    const licenseeData = { licensee: { firstName: 'Anikan', lastName: 'Skywalker' } }
+    const getCurrentPermission = jest.fn(() => samplePayload)
+    const mockRequest = createRequestMock(getCurrentPermission, licenseeData)
+    await updateTransaction(mockRequest)
+    expect(getCurrentPermission).toHaveBeenCalledWith(NAME.page)
+  })
```

Comment:
This test is more complicated than it needs to be: all you're testing is that `request.cache().helpers.page.getCurrentPermission` is called with `NAME.page`. Using the mock request creation function I've given above, you can do that in three lines:
```
const getPageData = jest.fn()
await updateTransaction(createRequestMock({ getPageData })
expect(getPageData).toHaveBeenCalledWith(NAME.page)
```